File info tool build errors with MPISERIAL

One cime_developer test (SMS_D_Ln9_Mmpi-serial.f19_g16_rx1.A) that uses MPISERIAL has some build errors from scorpio's file info tool.

Steps to reproduce this issue on anlworkstation (for MPISERIAL_PATH, mct directory inside cime bld directory can be used):
```
source /software/common/adm/packages/softenv-1.6.2/etc/softenv-load.sh
source /software/common/adm/packages/softenv-1.6.2/etc/softenv-aliases.sh
soft add +cmake-3.12.4
soft add +gcc-8.2.0
git clone https://github.com/E3SM-Project/scorpio.git
cd scorpio
mkdir build
cd build
CC=gcc CXX=g++ FC=gfortran cmake -DNetCDF_PATH=/soft/apps/packages/climate/netcdf/4.4.1c-4.2cxx-4.4.4f-serial/gcc-8.2.0 -DPIO_USE_MPISERIAL=ON -D MPISERIAL_PATH=path/to/mpiserial -DPIO_USE_MALLOC=ON -DWITH_PNETCDF=OFF -DPIO_ENABLE_TESTS=ON ..
make
```
The build errors are:
```
tools/util/spio_misc_tool_utils.cxx:30:47: error: 'MPI_C_BOOL' was not declared in this scope
     mpierr = MPI_Allreduce(&lsucc, &gsucc, 1, MPI_C_BOOL, MPI_LAND, comm);
                                               ^~~~~~~~~~
tools/util/spio_misc_tool_utils.cxx:30:47: note: suggested alternative: 'MPI_BXOR'
     mpierr = MPI_Allreduce(&lsucc, &gsucc, 1, MPI_C_BOOL, MPI_LAND, comm);
                                               ^~~~~~~~~~
                                               MPI_BXOR
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -33,6 +33,18 @@ option (WITH_NETCDF          "Require the use of NetCDF"                    ON)
 option (WITH_ADIOS2          "Require the use of ADIOS 2.x"                 OFF)
 option (ADIOS_BP2NC_TEST     "Enable testing of BP to NetCDF conversion"    OFF)
 
+# The tools and examples do not build with the MPI serial library
+if(PIO_USE_MPISERIAL)
+  if(PIO_ENABLE_TOOLS)
+    message(WARNING "Scorpio tools are not supported by the MPI serial library, disabling tools")
+    set(PIO_ENABLE_TOOLS 0)
+  endif()
+  if(PIO_ENABLE_EXAMPLES)
+    message(WARNING "Scorpio examples are not supported by the MPI serial library, disabling building examples")
+    set(PIO_ENABLE_EXAMPLES 0)
+  endif()
+endif()
+
 # Set a variable that appears in the pio_config.h.in file.
 if(PIO_USE_MALLOC)
   set(USE_MALLOC 1)

